webclient/pas/migrations/0007_remove_samostatnynalez_samostatny_nalez_geom_check_and_more.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 5.1.2 on 2024-11-04 20:43
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('core', '0016_alter_customadminsettings_options_and_more'),
+        ('heslar', '0007_alter_heslar_unique_together_and_more'),
+        ('historie', '0009_alter_historie_typ_zmeny'),
+        ('pas', '0006_views'),
+        ('projekt', '0008_alter_projekt_datum_ukonceni_and_more'),
+        ('uzivatel', '0016_organizace_cteni_dokumentu'),
+    ]
+
+    operations = [
+        migrations.RemoveConstraint(
+            model_name='samostatnynalez',
+            name='samostatny_nalez_geom_check',
+        ),
+        migrations.AlterField(
+            model_name='samostatnynalez',
+            name='geom_system',
+            field=models.TextField(default='5514'),
+        ),
+        migrations.AlterField(
+            model_name='samostatnynalez',
+            name='projekt',
+            field=models.ForeignKey(db_column='projekt', limit_choices_to={'typ_projektu': 1137}, on_delete=django.db.models.deletion.RESTRICT, related_name='samostatne_nalezy', to='projekt.projekt'),
```

Comment:
@jhavrlant Tohle vidím vyloženě jako problém, protože se tam typ projektu řídí ID z DB, což ale může být různé na různých serverech.

Reply:
@motyc Tak máme 3 možnosti jak to řešit. 

1. Nedělat nic, protože ten příkaz v databázi nic nedělá. Tady je výpis co ten soubor dělá v databázi: 
BEGIN;
--
-- Remove constraint samostatny_nalez_geom_check from model samostatnynalez
--
ALTER TABLE "samostatny_nalez" DROP CONSTRAINT "samostatny_nalez_geom_check";
--
-- Alter field geom_system on samostatnynalez
--
-- (no-op)
--
-- Alter field projekt on samostatnynalez
--
-- (no-op)
--
-- Create constraint samostatny_nalez_geom_check on model samostatnynalez
--
ALTER TABLE "samostatny_nalez" ADD CONSTRAINT "samostatny_nalez_geom_check" CHECK ((("geom_system" = '5514' AND "geom_sjtsk" IS NOT NULL) OR ("geom_system" = '4326' AND "geom" IS NOT NULL) OR ("geom_sjtsk" IS NULL AND "geom" IS NULL)));
COMMIT;

**2.** Smazat ten řádek v migraci, ale pak se nám bude generovat v těch následujících.
**3.** Ošetřit to v kódu django a udělat z té podmínky dynamickou a v migraci to potom bude vypadat takto:
 field=models.ForeignKey(db_column='projekt', limit_choices_to=pas.models.get_limit_choices, on_delete=django.db.models.deletion.RESTRICT, related_name='samostatne_nalezy', to='projekt.projekt'),
vliv na databázi opět žádný
**Přikláněl bych se k 1. Ty záznamy v migraci nejsou jen pokyny pro změny v databázi, ale také záznam změn v modelech django a proto ne všechno co se v migraci objeví se aplikuje v databázi.**

Reply:
Bude řešeno doplněním komentáře k danému řádku migrace.